src/datadog/grok/grok.rs

```diff
@@ -107,10 +110,20 @@ impl Pattern {
 
     /// Matches this compiled `Pattern` against the text and returns the matches.
     #[inline]
-    pub fn match_against<'a>(&'a self, text: &'a str) -> Option<Matches<'a>> {
-        self.regex
-            .captures(text)
-            .map(|cap| Matches::new(cap, &self.names))
+    pub fn match_against<'a>(
+        &'a self,
+        text: &'a str,
+    ) -> Result<Option<Matches<'a>>, GrokRuntimeError> {
+        let result = panic::catch_unwind(|| self.regex.captures(text));
+
+        match result {
+            Ok(Some(cap)) => Ok(Some(Matches::new(cap, &self.names))),
+            Ok(None) => Ok(None),
+            // https://github.com/rust-onig/rust-onig/issues/178
+            Err(_) => Err(GrokRuntimeError::FailedToMatch(
```

Comment:
Notes:
* This will match all panics but this block assumes it's always due to https://github.com/rust-onig/rust-onig/issues/178. 
* However, handling the `Err` payload is problematic ⚠️ , see last paragraph https://doc.rust-lang.org/std/panic/fn.catch_unwind.html#notes.
* I think this will work as is but flagging this for awareness.

Reply:
that's a good point, I've made the error message less specific for the time being: https://github.com/vectordotdev/vrl/pull/1079/commits/3acf0d08bac697b33331fe4151189060fbc90c94 

since the panic still gets printed to console, users should hopefully have enough indication of what went wrong e.g. for this issue the panic would show 
```
Regex search error: retry-limit-in-match over
```